### PR TITLE
JDK-8305118: Add RISC-V related content to building.md

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch adds the RISC-V related content to `building.md`. But I didn't 
generate `building.html` locally, because I know the `pandoc` version 
may influencethe the output of `building.html` (Need help here).

I know there are many steps and subtle points about cross-compiling the 
external libraries. Considering the content length and the libraries are 
introduced at `External Library Requirements`, I give only one example 
instead of all of them.

And I tried to build it by using `devkits` and `Debian sysroots`. But it failed locally.
Maybe someone who has built successfully can tell me how to change the
document, which mainly adds the right name to their supported target table.

Thanks for the review and assist.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305118](https://bugs.openjdk.org/browse/JDK-8305118): Add RISC-V related content to building.md

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13223/head:pull/13223` \
`$ git checkout pull/13223`

Update a local copy of the PR: \
`$ git checkout pull/13223` \
`$ git pull https://git.openjdk.org/jdk.git pull/13223/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13223`

View PR using the GUI difftool: \
`$ git pr show -t 13223`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13223.diff">https://git.openjdk.org/jdk/pull/13223.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13223#issuecomment-1488334080)

### CODE REVIEW

**Comment by shipilev** [comment]
RISC-V cross-compiles with sysroots fine:
https://builds.shipilev.net/openjdk-jdk/configure-logs/configure-openjdk-jdk-linux-riscv64-server-release-gcc10-glibc2.31.log
https://builds.shipilev.net/openjdk-jdk/build-logs/build-openjdk-jdk-linux-riscv64-server-release-gcc10-glibc2.31.log

Sometimes you need to point to tools like `OBJCOPY` and `STRIP`, if the host/target triplets contain `unknown`. But other than that, it should be pretty straight-forward.

File: doc/building.md [hunk 1]
```diff
@@ -1226,6 +1226,42 @@ available using `--with-abi-profile`: arm-vfp-sflt, arm-vfp-hflt, arm-sflt,
...
+    git clone --recursive https://github.com/riscv-collab/riscv-gnu-toolchain
+    cd riscv-gnu-toolchain
+    ./configure --prefix=<toolchain-installed-path>
+    make linux
+    export PATH=<toolchain-installed-path>/bin:$PATH
+    ```
+
+  * Cross-compile all the required libraries:
+    ```
+    # An example about libffi
```
Comment (erikj79) [inline doc/building.md:1246]:
```suggestion
    # An example for libffi
```

File: doc/building.md [hunk 2]
```diff
@@ -1226,6 +1226,42 @@ available using `--with-abi-profile`: arm-vfp-sflt, arm-vfp-hflt, arm-sflt,
 armv5-vfp-sflt, armv6-vfp-hflt. Note that soft-float ABIs are no longer
 properly supported by the JDK.
 
+### Building for RISC-V
+
+The RISC-V community provides a basic [GNU compiler toolchain](https://github.com/riscv-collab/riscv-gnu-toolchain).
+But the [external libraries](#External-Library-Requirements) required by OpenJDK complicate the building process.
+The placeholder `<toolchain-installed-path>` shown below is the path you want to install the toolchain.
```
Comment (erikj79) [inline doc/building.md:1233]: Please keep paragraph text to 80 character width.

"is the path you want" -> "is the path where you want"

**Review by erikj79 — COMMENTED** [review]
Just some formatting and grammar notes. I can help you with running pandoc later.

File: doc/building.md (hunk 1 above)
Comment (lgxbslgx) [reply doc/building.md]: Fixed.

File: doc/building.md (hunk 2 above)
Comment (lgxbslgx) [reply doc/building.md]: Fixed.

**Comment by lgxbslgx** [comment]
> RISC-V cross-compiles with sysroots fine: https://builds.shipilev.net/openjdk-jdk/configure-logs/configure-openjdk-jdk-linux-riscv64-server-release-gcc10-glibc2.31.log https://builds.shipilev.net/openjdk-jdk/build-logs/build-openjdk-jdk-linux-riscv64-server-release-gcc10-glibc2.31.log
> 
> Sometimes you need to point to tools like `OBJCOPY` and `STRIP`, if the host/target triplets contain `unknown`. But other than that, it should be pretty straight-forward.

Thanks for the tips. I know why I failed locally now. The mirror path should be `debian-ports` instead of `debian`. `http://deb.debian.org/debian-ports` and `http://httpredir.debian.org/debian-ports` succeed. The file [.github/workflows/build-cross-compile.yml](https://github.com/openjdk/jdk/blob/b3ff8d1c89b0f968b7b5ec2105502778524e4e4a/.github/workflows/build-cross-compile.yml#L89) provides the concrete information.

File: doc/building.md [hunk 3]
```diff
@@ -1226,6 +1230,44 @@ available using `--with-abi-profile`: arm-vfp-sflt, arm-vfp-hflt, arm-sflt,
 armv5-vfp-sflt, armv6-vfp-hflt. Note that soft-float ABIs are no longer
 properly supported by the JDK.
 
+### Building for RISC-V
+
+The RISC-V community provides a basic
+[GNU compiler toolchain](https://github.com/riscv-collab/riscv-gnu-toolchain).
+But the [external libraries](#External-Library-Requirements) required by OpenJDK
```
Comment (liach) [inline doc/building.md:1237]:
```suggestion
However, the [external libraries](#External-Library-Requirements) required by OpenJDK
```
or change 'But' to ', but'

Comment (lgxbslgx) [reply doc/building.md]: Fixed.

**Comment by erikj79** [comment]
Generated html file here https://github.com/erikj79/jdk/commits/pull/13223

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**